esphome/components/audio_dac/audio_dac.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/core/defines.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace audio_dac {
+
+class AudioDac : public Component {
+ public:
+  virtual void set_mute_off(){ this->is_muted = false; }
+  virtual void set_mute_on(){ this->is_muted = true; }
+  virtual void set_volume(optional<float> volume){}
+
```

Comment:
Could you add ``get_volume`` and ``get_mute`` (or ``get_mute_state`` maybe would be a better name?) virtual functions? It's good to read these after a software reset so the ESP32 can know what the existing DAC states are. 

For the AIC3204, you just read the respective registers instead of write (plus a map back to a float for the volume)

Reply:
[`f45ef37` (#59)](https://github.com/esphome/voice-kit/pull/59/commits/f45ef37772638d10b6e18f519cbe1fd92268a665) -- I did this the "lazy" way. Instead of reading back registers from the DAC (which takes a long time...and they should not ever change unless we change them), the methods simply return stored values, and I updated the init process in `setup()` to write the specified values rather than fixed defaults.

Reply:
Ah that's right, initializing the DAC will erase the existing settings. It would be nice to still read the current mute and volume from the DAC before the initialization procedure and set them to that during initialization. That way if someone does an OTA update, the mute and volume settings are still retained after the reset.

Reply:
I think the best way to do that would be to store the settings in ESPHome preferences and just reset them on the DAC when it's initialized.

Reply:
Since we discussed in our meeting -- marking resolved. 😇